test/tests/functional/pbs_acl_groups.py

```diff
@@ -0,0 +1,80 @@
+# coding: utf-8
+
+# Copyright (C) 1994-2017 Altair Engineering, Inc.
+# For more information, contact Altair at www.altair.com.
+#
+# This file is part of the PBS Professional ("PBS Pro") software.
+#
+# Open Source License Information:
+#
+# PBS Pro is free software. You can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
+# details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# The PBS Pro software is licensed under the terms of the GNU Affero General
+# Public License agreement ("AGPL"), except where a separate commercial license
+# agreement for PBS Pro version 14 or later has been executed in writing with
+# Altair.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software - under
+# a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+from tests.functional import *
+
+
+class Test_acl_groups(TestFunctional):
+    """
+    Test to check acl_groups and acl_resv_groups considers secondary group
+    """
+
+    def test_acl_grp_queue(self):
+        """
+        Set acl_groups on a queue and submit a job with a user
+        for whom the set group is a secondary group
+        """
+        a = {'queue_type': 'execution', 'started': 't', 'enabled': 't',
+             'acl_group_enable': 't', 'acl_groups': TSTGRP1}
+        self.server.manager(MGR_CMD_CREATE, QUEUE, a, id='workq2')
+        a = {'queue': 'workq2'}
+        j = Job(TEST_USER1, attrs=a)
+        try:
```

Comment:
Please add a comment here to explain that the submit should not error out for the test to pass, it's not obvious here that the submit failing is the failure scenario for the test case as the except is doing an assert.